sql/parser/builtins.go

```diff
@@ -116,6 +119,181 @@ var builtins = map[string][]builtin{
 		},
 	},
 
+	"concat_ws": {
+		builtin{
+			fn: func(args DTuple) (Datum, error) {
+				dstr, ok := args[0].(DString)
+				if !ok {
+					return DNull, fmt.Errorf("unknown signature for concat_ws: concat_ws(%s, ...)", args[0].Type())
+				}
+				sep := string(dstr)
+				var ss []string
+				for _, d := range args[1:] {
+					if d == DNull {
+						continue
+					}
+					ds, err := datumToRawString(d)
+					if err != nil {
+						return DNull, err
+					}
+					ss = append(ss, ds)
+				}
+				return DString(strings.Join(ss, sep)), nil
+			},
+		},
+	},
+
+	"split_part": {
+		builtin{
+			types: typeList{stringType, stringType, intType},
+			fn: func(args DTuple) (Datum, error) {
+				text := string(args[0].(DString))
+				sep := string(args[1].(DString))
+				field := int(args[2].(DInt))
+
+				if field <= 0 {
+					return DNull, fmt.Errorf("field position must be greater than zero")
```

Comment:
(1) this error should mention the function's name; this is important when a single statement calls multiple functions
(2) can this error message be static? i.e. `errFieldPositionGtZero = errors.new(...)`